src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -272,17 +271,17 @@ public object ResolveList(JsonElement array, IObjectField fieldSchema, ref IMeta
             }
             else if (array.ValueKind is JsonValueKind.String)
             {
-                using ArrayPoolWriter buffer = new();
```

Comment:
Instead of removing usage of ArrayPoolWriter altogether, can we fix the memory advancement bugs in its implementation?
ArrayPoolWriter was introduced specifically to avoid usage of JsonDocument.Parse.. Reintroducing JsonDocument, seems like we are going back in time.

If we indeed end up removing ArrayPoolWriter usage, we should get rid of its implementation as well from our project.

Reply:
This was what i was trying before, used different approaches like increasing neededCapacity, but this was again uncertain as to how much would be correct as it was still giving uncertain results. And it was harder to debug as well due to manual buffer management and encoding conversion as to what exactly is causing this issue.

That's why i resorted to JsonDocument to reduce the complexity of this process.

Reply:
can do a clean up in a separate PR